include/RAJA/pattern/kernel/For.hpp

```diff
@@ -101,7 +101,8 @@ struct StatementExecutor<
     auto len = segment_length<ArgumentId>(data);
     using len_t = decltype(len);
 
-    auto r = data.res;
```

Comment:
Does this change remain necessary? The resource passed in here should be the right one when the user is passing one in, and we probably don't want to disable that.  Will the rest work if this change is pulled out?

Reply:
Changing it back breaks the following example:
```
Policy = KernelPolicy<
  For<0,seq,
    For<1,omp_target
      Lambda<0>>>>;

kernel<Policy>(...) // no resource specified
```

**The issue:**

kernel<Policy> creates a default resource based on Policy. 
Default resource creates the default based on get_platform<Policy>. 
This should basically resolve to platform_max(omp_target, host) = omp_target.
Default resource for omp_target is resource::Omp
So we have resource::Omp being passed to the LoopData constructor and then used throughout the execution.

When we get to the first StatementExecutor, which is for seq_exec, we're gonna call forall_impl for a seq_exec and resource::Omp. No such forall_impl exists.

**Alternative change:** 
Implement a forall_impl for a seq_exec policy and a omp resource. 

Do you anticipate implementing such a forall_impl to cause any other issues?